rawx-lib/src/attr_handler.c

```diff
@@ -265,6 +249,63 @@ get_rawx_info_from_fd(int fd, GError **error, gchar *hex_chunkid,
 	}
 	g_free(attr_name_content_fullpath);
 
+	return rc;
+}
+
+gboolean
+get_rawx_fullpath_info_from_file(const char *p, GError **error, gchar *hex_chunkid,
+		struct chunk_textinfo_s *chunk)
+{
+	int fd = open(p, O_RDONLY);
+	if (fd < 0) {
+		GSETCODE(error, errno, "open() error: (%d) %s", errno, strerror(errno));
+		return FALSE;
+	} else {
+		gboolean rc = get_rawx_fullpath_info_from_fd(fd, error, hex_chunkid, chunk);
+		int errsav = errno;
+		metautils_pclose (&fd);
+		errno = errsav;
+		return rc;
+	}
+}
+
+gboolean
+get_rawx_info_from_fd(int fd, GError **error, gchar *hex_chunkid,
+		struct chunk_textinfo_s *chunk)
+{
+	if (fd < 0) {
+		GSETCODE(error, EINVAL, "invalid FD");
+		return FALSE;
+	}
+	if (!hex_chunkid) {
+		GSETCODE(error, EINVAL, "invalid chunk ID");
+		return FALSE;
+	}
+
+	if (!chunk) {
+		gchar *v = NULL;
+		if (!GET(ATTR_NAME_CONTENT_STGPOL, v)) {
+			if (errno == ENOTSUP) {
+				GSETCODE(error, errno, "xattr not supported");
+				return FALSE;
+			}
+		} else {
+			g_free0 (v);
+		}
+		return TRUE;
+	}
+
+	if (!GET(ATTR_NAME_CONTENT_STGPOL, chunk->content_storage_policy)) {
```

Comment:
Since this operation is done either way, I suggest loading the value into `v`, and then either:
- copy the pointer into `chunk->content_storage_policy`;
- or free it.